solver/src/naive_solver.rs

```diff
@@ -19,17 +20,11 @@ impl Solver for NaiveSolver {
         _gas_price: f64,
     ) -> Result<Option<Settlement>> {
         let mut limit_orders = Vec::new();
-        let mut uniswaps = HashMap::new();
+        let uniswaps = extract_deepest_amm_liquidity(&liquidity);
         for liquidity in liquidity {
             match liquidity {
                 Liquidity::Limit(order) => limit_orders.push(order),
-                Liquidity::Amm(uniswap) => {
-                    let pair = uniswap.tokens;
-                    uniswaps.insert(
-                        TokenPair::new(pair.get().0, pair.get().1).expect("Invalid Pair"),
-                        uniswap,
-                    );
-                }
+                Liquidity::Amm(_) => continue,
             }
         }
```

Comment:
ubernit: Now that this code is purely for "extracting limit orders from liquidity", maybe something like:
```
let limit_orders = liquidity.iter.filter_map(|liquidity| match liquidity {
    Liquidity::Limit(order) => Some(order),
    _ => None,
}).collect();
```